subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/impl/legacy/LegacyGradleEnterprisePluginCheckInService.java

```diff
@@ -58,11 +58,16 @@ public LegacyGradleEnterprisePluginCheckInService(
         this.buildType = buildType;
     }
 
-    public static String unsupportedReason() {
+    @Nullable
+    private String unsupportedReason(VersionNumber pluginVersion) {
         if (Boolean.getBoolean(UNSUPPORTED_TOGGLE)) {
             return UNSUPPORTED_TOGGLE_MESSAGE;
+        } else if (gradle.getStartParameter().isConfigurationCache()) {
+            return "Build scans have been disabled due to incompatibility between your Gradle Enterprise plugin version (" + pluginVersion.toString() + ") and configuration caching. " +
```

Comment:
Could you also update the 'build scan support' section in the configuration cache user manual chapter? Currently it says that (only) `--scan` works, but I don't think this will be the case any more (which is fine, we should just keep the docs in sync with the behaviour).